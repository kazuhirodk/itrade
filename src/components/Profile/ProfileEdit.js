import React, { Component } from 'react';
import { Actions } from 'react-native-router-flux';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  Button
} from 'react-native';

const goToHome = () => {
  Actions.home()
}

export default class Profile extends Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}></View>
        <Image style={styles.avatar} source={require('../../components/images/avatar.png')}/>
        <View style={styles.body}>
          <Text style={styles.name}>Fox Developer</Text>
          <Text style={styles.info}>Mobile Developer / UX Designer</Text>
          <Text style={styles.description}>Lorem ipsum dolor sit amet, saepe sapientem eu nam. Qui ne assum electram expetendis, omittam deseruisse consequuntur ius an</Text>
        </View>
        <TextInput
          style={styles.input}
          placeholder='Nome'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
        />
        <TextInput
          style={styles.input}
          placeholder='Email'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
        />
        <TextInput
          style={styles.input}
          placeholder='Telefone'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
        />
        <Button
          color='#239033'
          title='Salvar dados'
          onPress={goToHome}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 20,
    flex: 1,
    backgroundColor: '#FFFAFA',
    alignContent: 'center'
  },
  header:{
    backgroundColor: "#FD6D64",
    height:200,
  },
  avatar: {
    width: 130,
    height: 130,
    borderRadius: 63,
    borderWidth: 4,
    borderColor: "white",
    marginBottom:10,
    alignSelf:'center',
    position: 'absolute',
    marginTop:130
  },
  body:{
    marginTop: 40,
    alignItems: 'center',
    marginBottom: 40
  },
  name:{
    fontSize:28,
    color: "#696969",
    fontWeight: "600"
  },
  info:{
    fontSize:16,
    color: "#CD7F32",
    marginTop:10
  },
  description:{
    fontSize:16,
    color: "#696969",
    marginTop:10,
    textAlign: 'center'
  },
  input: {
    height: 40,
    backgroundColor: '#E3E1E1',
    marginBottom: 10,
    padding: 10,
    color: '#000'
  },
});
