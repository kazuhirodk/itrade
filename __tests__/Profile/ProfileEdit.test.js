import ProfileEdit from '../../src/components/Profile/ProfileEdit.js'
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const wrapper = shallow(<ProfileEdit />);

describe('ProfileEdit', () => {
  it('matches snapshot properly', () => {
    const jsonProfileEdit = renderer.create(<ProfileEdit />).toJSON();
    expect(jsonProfileEdit).toMatchSnapshot();
  });

  describe('#onChangeText', () => {
    it('receive one or more info and add to state', () => {
      wrapper.instance().onChangeText('name', 'Nome do Usuário');
      wrapper.instance().onChangeText('email', 'email@teste.com');

      expect(wrapper.instance().state.name).toEqual('Nome do Usuário');
      expect(wrapper.instance().state.email).toEqual('email@teste.com');
    })
  });
});
