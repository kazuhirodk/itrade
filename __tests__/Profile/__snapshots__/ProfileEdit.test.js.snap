// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProfileEdit matchs snapshot properly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFAFA",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FD6D64",
        "height": 200,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/components/images/avatar.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "white",
        "borderRadius": 63,
        "borderWidth": 4,
        "height": 130,
        "marginBottom": 10,
        "marginTop": 130,
        "position": "absolute",
        "width": 130,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 40,
        "marginTop": 40,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Nome"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Telefone"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#239033",
            },
          ]
        }
      >
        Salvar dados
      </Text>
    </View>
  </View>
</View>
`;

exports[`ProfileEdit matchs snapshot properly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFAFA",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FD6D64",
        "height": 200,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/components/images/avatar.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "white",
        "borderRadius": 63,
        "borderWidth": 4,
        "height": 130,
        "marginBottom": 10,
        "marginTop": 130,
        "position": "absolute",
        "width": 130,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 40,
        "marginTop": 40,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Nome"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Telefone"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#239033",
            },
          ]
        }
      >
        Salvar dados
      </Text>
    </View>
  </View>
</View>
`;
